default/python/AI/AIFleetMission.py

```diff
@@ -564,6 +590,41 @@ def issue_fleet_orders(self):
                         MilitaryAI.assign_military_fleets_to_systems(use_fleet_id_list=new_military_fleets,
                                                                      allocations=allocations)
 
+    def _portion_of_fleet_needed_here(self):
+        """Calculate the portion of the fleet needed in target system considering enemy forces."""
+        # TODO check rating against planets
+        if assertion_fails(self.type in COMBAT_MISSION_TYPES):
+            error("%s" % self)
```

Comment:
Can be simplified to `error(self)`.

You can add mesage to check.
`assertion_fails(self.type in COMBAT_MISSION_TYPES, str(self)):`